Convert mtime to int before comparing it

The st_mtime attribute (which is a float) is compared against a value
from the timestamp database, which was stored as an integer there.

When working on a filesystem with precise timestamps the comparision
will fail nearly everytime hence.

Although it might be possible to enhance the database to store the
fractional part too, this will complicate things more than we would
gain by this change.

Signed-off-by: Enrico Scholz <enrico.scholz@sigma-chemnitz.de>

Upstream-Status: Pending
(Contacting the original author, no response yet.)

Index: opkg-utils/opkg-make-index
===================================================================
--- opkg-utils.orig/opkg-make-index
+++ opkg-utils/opkg-make-index
@@ -100,7 +100,7 @@ for filename in files:
      pkg = None
      fnameStat = os.stat(filename)
      if old_pkg_hash.has_key(basename):
-          if pkgsStamps.has_key(basename) and fnameStat.st_mtime == pkgsStamps[basename]:
+          if pkgsStamps.has_key(basename) and int(fnameStat.st_mtime) == pkgsStamps[basename]:
             if (verbose):
                sys.stderr.write("Found %s in Packages\n" % (filename,))
             pkg = old_pkg_hash[basename]
